### ISSUE TITLE
Pyright fails to report unused protected methods

### ISSUE THREAD

**Issue by acampove** [microsoft/pyright#11137]
**Describe the bug**

Pyright seems  to only report unused private methods but not unused protected ones. This is a problem, because no one uses private methods in python.

**Code or Screenshots**

<img width="860" height="221" alt="Image" src="https://github.com/user-attachments/assets/fd906586-6d2f-4a06-bfb7-d3ca78e903f8" />

```python
class MyClass:
    def _method(self):
        pass

    def __method(self):
        pass
```

**VS Code extension or command-line**

I am using neovim

**Comment by erictraut** [comment]
Pyright is working as designed here, so this isn't a bug.

**Comment by erictraut** [comment]
This rule is arguably falls more into the category of linting rather than type checking. If I were creating pyright from scratch today, I'd probably not include this check. If you prefer different linting rules for your project, I recommend disabling this check in pyright and using a linter like ruff or pylint to enforce rules like this.

**Comment by acampove** [comment]
> This rule is arguably falls more into the category of linting rather than type checking. If I were creating pyright from scratch today, I'd probably not include this check. If you prefer different linting rules for your project, I recommend disabling this check in pyright and using a linter like ruff or pylint to enforce rules like this.

Ruff does not seem to do this check either. I see other tools that seem to do it, but they are maintained by just one person and the code and documentation behind those tools is not very good, so I am not going to use them. Right now, using this unintended feature of pyright seems to be the only way to identify dead code. However this only works if I make all my methods private. 

Despite all the talk out there, in forums, about not wanting methods to be private; a method should be protected if it is intended to be reimplemented in a derived class. Most of my protected methods (as is the case in most code out there) will never be reimplemented in a subclass, subclassing itself is something taht I do for maybe less than 5% of my classes. So having protected methods everywhere instead of private ones does not make much sense anyway. 

i would rather have most of my non-public methods as private rather than thousands of lines of dead code that I am not aware of.
